### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jameslibscie**
Very curious whether I've done this right and whether you agree with these guidelines. I've made some educated guesses here and there, but could be completely off on some things since open source dev is new to me.

I thought I'd start with the SDK first, as that seems the most logical place for outside contributors. Next up would be Hypergraph CLI, Daemon app, Desktop in that order. Does that make sense? They would be similar, but contain some product-specific stuff (like links to Figma for Desktop).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+# Contributing to @p2pcommons/sdk-js
+
+Welcome! We *love* that you're interested in contributing to p2pcommons :purple_heart:
+
+We recognize all our contributors following the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
+
+Come and join us on [Slack](https://libscie.slack.com/) to chat with our team and stay up to date with all [Liberate Science](https://github.com/libscie) goings-on.
```
Comment (chartgerink) [inline CONTRIBUTING.md]:
```suggestion
Come and join us on [Slack](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw) to chat with our team and stay up to date with all [Liberate Science](https://libscie.org) goings-on.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Contributing to @p2pcommons/sdk-js
+
+Welcome! We *love* that you're interested in contributing to p2pcommons :purple_heart:
+
+We recognize all our contributors following the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
```
Comment (chartgerink) [inline CONTRIBUTING.md]:
```suggestion
We recognize all our contributors using the [all-contributors](https://github.com/all-contributors/all-contributors) bot. Contributions of any kind welcome!
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Contributing code
+
+### Where to start?
+
+Our work is organized on our [GitHub Projects board](https://github.com/orgs/p2pcommons/projects/1). You can help us the most by contributing to issues in the "Backlog" column. These are issues that are already specified and considered ready to start work on.
+
+The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/).
+
+Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome. Please create an issue first, but feel free to start work right away.
```
Comment (chartgerink) [inline CONTRIBUTING.md]:
```suggestion
Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## Code of conduct
+
+Please note we adhere to a [Code of Conduct](https://github.com/hypergraph-xyz/cli/blob/master/CODE_OF_CONDUCT.md) and any contributions not in line with it (*tl;dr* be an empathetic, considerate person) will not be accepted. Please notify [@chartgerink](mailto:chris@libscie.org) if anything happens.
+
+## Feature requests & bug reports
+
+Feature requests and bug reports can be submitted through [GitHub issues](https://github.com/p2pcommons/sdk-js/issues). If the request concerns a proposal for a change to the module specifications, please submit it to our [specs repo](https://github.com/p2pcommons/specs/issues).
+
+Templates are offered for feature requests and bug reports. These are not mandatory, but preferred.
```
Comment (chartgerink) [inline CONTRIBUTING.md]:
```suggestion
We offer templates for feature requests and bug reports.
```

Active voice :)

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/).
+
+Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome. Please create an issue first, but feel free to start work right away.
+
+***Please note that we have [specs](https://github.com/p2pcommons/specs/issues) that the application must adhere to. If you believe these specs need amending, submit an issue to that repo.***
+
+### Git guidelines
+
+When working on an issue, self-assign and switch the `ready to start` label for `in progress`. Please always create a new branch for your work and submit a pull request that closes the issue when done. If you'd like input along the way, a draft pull request is also fine. Please invite recent contributors to review (at least one, but preferably two or more). Project maintainers will take care of releases.
```
Comment (chartgerink) [inline CONTRIBUTING.md]: This is only possible for members of the repository. 

Repository maintainers should sign their commits with their PGP key.

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+Please report security vulnerabilities directly to [@chartgerink](mailto:chris@libscie.org).
+
+## Contributing code
+
+### Where to start?
+
+Our work is organized on our [GitHub Projects board](https://github.com/orgs/p2pcommons/projects/1). You can help us the most by contributing to issues in the "Backlog" column. These are issues that are already specified and considered ready to start work on.
+
+The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/).
```
Comment (chartgerink) [inline CONTRIBUTING.md]:
```suggestion
The "PM Inbox" column and our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contain issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/) ([invite link](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw)).
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+Welcome! We *love* that you're interested in contributing to p2pcommons :purple_heart:
+
+We recognize all our contributors following the [all-contributors](https://github.com/all-contributors/all-contributors) specification. Contributions of any kind welcome!
+
+Come and join us on [Slack](https://libscie.slack.com/) to chat with our team and stay up to date with all [Liberate Science](https://github.com/libscie) goings-on.
+
+## Code of conduct
+
+Please note we adhere to a [Code of Conduct](https://github.com/hypergraph-xyz/cli/blob/master/CODE_OF_CONDUCT.md) and any contributions not in line with it (*tl;dr* be an empathetic, considerate person) will not be accepted. Please notify [@chartgerink](mailto:chris@libscie.org) if anything happens.
```
Comment (juliangruber) [inline CONTRIBUTING.md]: can we move the code of conduct to a central place, instead of linking to the cli repo?

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+### Security vulnerabilities
+
+Please report security vulnerabilities directly to [@chartgerink](mailto:chris@libscie.org).
+
+## Contributing code
+
+### Where to start?
+
+Our work is organized on our [GitHub Projects board](https://github.com/orgs/p2pcommons/projects/1). You can help us the most by contributing to issues in the "Backlog" column. These are issues that are already specified and considered ready to start work on.
```
Comment (juliangruber) [inline CONTRIBUTING.md]: I don't think anyone should need to know about our column system. People can simply report issues and open PRs, we will triage them as we see fit. If an outsider wants to work on an issue, we can't assign it to them anyway, unless they get added to the project. Usually people can just comment that they want to work on this.

File: CONTRIBUTING.md (hunk 3 above)
Comment (juliangruber) [reply CONTRIBUTING.md]: Agreed, at this stage _any_ contribution is better than following a process

**Comment by dpaez** [comment]
> I don't know why @dpaez his commits are in this PR, hihi?

u.u 
Sorry for the extra files on the PR, I accidentally added those when I update the CONTRIBUTING file. Anyway, I guess that we can merge the PR normally when we finish with the review ✔️

File: CONTRIBUTING.md (hunk 5 above)
Comment (jameslibscie) [reply CONTRIBUTING.md]: Are you referring to the labels or also the branching & PR? Would it make sense to use forks instead @dpaez @juliangruber ?

Comment (dpaez) [reply CONTRIBUTING.md]: external contributors will need to fork to contribute. We can help them with labels if needed.

File: CONTRIBUTING.md (hunk 8 above)
Comment (dpaez) [reply CONTRIBUTING.md]: I like those changes

Comment (juliangruber) [reply CONTRIBUTING.md]: Sounds great!

File: CONTRIBUTING.md (hunk 7 above)
Comment (jameslibscie) [reply CONTRIBUTING.md:11]: I suggest we move it to the website when it's up. @chartgerink - any ideas?

File: CONTRIBUTING.md (hunk 8 above)
Comment (jameslibscie) [reply CONTRIBUTING.md]: @dpaez @juliangruber Update made!

File: CONTRIBUTING.md (hunk 5 above)
Comment (jameslibscie) [reply CONTRIBUTING.md]: @dpaez Update made! Please make or suggest changes to the forking stuff if necessary :)

Comment (dpaez) [reply CONTRIBUTING.md]: updated

File: CONTRIBUTING.md (hunk 7 above)
Comment (dpaez) [reply CONTRIBUTING.md:11]: another option is to create a repo with code of conduct inside one of the orgs and link everything to that repo

Comment (jameslibscie) [reply CONTRIBUTING.md:11]: Would you suggest a separate repo or perhaps an @libscie/docs repo with code of conduct, maybe glossary and perhaps even the business plan and manifesto (when that's done)?

File: CONTRIBUTING.md (hunk 5 above)
Comment (jameslibscie) [reply CONTRIBUTING.md]: Nice!

File: CONTRIBUTING.md (hunk 7 above)
Comment (chartgerink) [reply CONTRIBUTING.md:11]: I'm used to putting a code of conduct in each repo, but I think it makes sense to make that part of the libscie website. Could you leave it like this and I'll get on that this/next week? :) @juliangruber would that be okay with you too?

**Comment by juliangruber** [comment]
@dpaez could you rebase this PR to only include the contributing.md?

**Comment by dpaez** [comment]
I did the rebase but it seems like it didn't help :/

**Comment by juliangruber** [comment]
ok let me give this a try

**Comment by juliangruber** [comment]
this is a github issue it seems, as `git diff` is clean

**Comment by juliangruber** [comment]
Here is the raw diff:

```diff
diff --git a/CHANGELOG.md b/CHANGELOG.md
index b60afc5..48d3796 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - new deleteFiles option to delete method #90
 - new check of modified times at ready #138
+- new contributing doc
 
 ### Fixed
 - add missing check on main field (publish) #141
diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
new file mode 100644
index 0000000..8abe993
--- /dev/null
+++ b/CONTRIBUTING.md
@@ -0,0 +1,109 @@
+# Contributing to @p2pcommons/sdk-js
+
+Welcome! We *love* that you're interested in contributing to p2pcommons :purple_heart:
+
+We recognize all our contributors using the [all-contributors](https://github.com/all-contributors/all-contributors) bot. Contributions of any kind welcome!
+
+Come and join us on [Slack](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw) to chat with our team and stay up to date with all [Liberate Science](https://libscie.org) goings-on.
+
+## Code of conduct
+
+Please note we adhere to a [Code of Conduct](https://github.com/hypergraph-xyz/cli/blob/master/CODE_OF_CONDUCT.md) and any contributions not in line with it (*tl;dr* be an empathetic, considerate person) will not be accepted. Please notify [@chartgerink](mailto:chris@libscie.org) if anything happens.
+
+## Feature requests & bug reports
+
+Feature requests and bug reports can be submitted through [GitHub issues](https://github.com/p2pcommons/sdk-js/issues). If the request concerns a proposal for a change to the module specifications, please submit it to our [specs repo](https://github.com/p2pcommons/specs/issues).
+
+We offer templates for feature requests and bug reports.
+
+### Security vulnerabilities
+
+Please report security vulnerabilities directly to [@chartgerink](mailto:chris@libscie.org).
+
+## Contributing code
+
+### Where to start?
+
+Our work is organized on [GitHub Issues](https://github.com/p2pcommons/sdk-js/issues). Our [Roadmap](https://github.com/p2pcommons/sdk-js/wiki/Roadmap) contains issues that we are planning on working on further down the line. If you're enthusiastic about one of these features, come and discuss it with us on [Slack](https://libscie.slack.com/) ([invite link](https://join.slack.com/t/libscie/shared_invite/zt-9l0ig1x1-Sxjun7D6056cOUQ2Ai_Bkw)).
+
+Technical improvements, bug fixes, documentation and other non-feature work is also totally welcome.
+
+***Please note that we have [specs](https://github.com/p2pcommons/specs/issues) that the application must adhere to. If you believe these specs need amending, submit an issue to that repo.***
+
+### Git guidelines
+
+When starting work on an issue, please comment to say you're working on it. Create a fork for your work and submit a pull request that closes the issue when done. If you'd like input along the way, a draft pull request is also fine. Please invite recent contributors to review (at least one, but preferably two or more). Project maintainers will take care of releases and labels, etc. Feel free to ask questions at any time.
+
+In general the ideal PR process looks like this (just for reference):
+
+#### Step 1: Fork
+
+Fork the project [on GitHub](https://github.com/p2pcommons/sdk-js) and check out your copy locally.
+
+```bash
+$ git clone git@github.com:username/sdk-js.git
+$ cd sdk-js
+$ npm install
+$ git remote add upstream git://github.com/p2pcommons/sdk-js.git
+```
+
+#### Step 2: Branch
+
+Create a feature branch and start hacking:
+
+```bash
+$ git checkout -b my-feature-branch -t origin/master
+```
+
+#### Step 3: Test
+
+Bug fixes and features **should come with tests**. 
+
+```bash
+$ npm test
+```
+
+#### Step 4: Lint
+
+Make sure the linter is happy and that all tests pass. Please, do not submit
+patches that fail either check.
+
+We use [standard](https://standardjs.com/)
+
+#### Step 5: Commit
+
+Make sure git knows your name and email address:
+
+```bash
+$ git config --global user.name "Bruce Wayne"
+$ git config --global user.email "bruce@batman.com"
+```
+
+Writing good commit logs is important. A commit log should describe what
+changed and why.
+
+#### Step 6: Changelog
+
+If your changes are really important for the project probably the users want to know about it.
+
+We use [chan](https://github.com/geut/chan/) to maintain a well readable changelog for our users.
+
+#### Step 7: Push
+
+```bash
+$ git push origin my-feature-branch
+```
+
+#### Step 8: Make a pull request ;)
+
+### Documentation
+
+If your work adds functionality or changes the way existing functionality works, please document this in the [README.md](https://github.com/p2pcommons/sdk-js/blob/master/README.md). Always add changes to the [CHANGELOG.md](https://github.com/p2pcommons/sdk-js/blob/master/CHANGELOG.md).
+
+**Tip**: We are using [`chan`](https://github.com/geut/chan/tree/master/packages/chan) to easily modify and manage the changelog.
+
+### Style guide
+
+We are using `eslint` with the `standard` [config](https://github.com/standard/eslint-config-standard).
+
+**Tip**: If you are using an editor like `vim` you can check out GEUT's [xd](https://github.com/geut/xd) to improve your _dx_. :v:

```

**Pull request merged**